ship_functions.py

```diff
@@ -0,0 +1,212 @@
+                                model_dir=logs_path)
+
+    #Exclude the last COCO layers because they require a matching number of classes
+    if weights.lower() == "coco":
+        ship_model.load_weights(weights_path, by_name=True, exclude=[
+            "mrcnn_class_logits", "mrcnn_bbox_fc",
+            "mrcnn_bbox", "mrcnn_mask"])
+    else:
+        ship_model.load_weights(weights_path, by_name=True)
+
+    return ship_model, config
+
+
+def launch_training(dataset_dir, train, val, local_model, config):
+    '''Launch Mask RCNN training'''
+    # Training dataset.
+    dataset_train = ship_dataset.ShipDataset()
+    dataset_train.load_ship(dataset_dir + "train/", train)
+    dataset_train.prepare()
+
+    # Validation dataset
+    dataset_val = ship_dataset.ShipDataset()
+    dataset_val.load_ship(dataset_dir + "val/", val)
+    dataset_val.prepare()
+
+    # Beginning of the training
+    print("Training network heads")
+    local_model.train(dataset_train, dataset_val,
+                      learning_rate=config.LEARNING_RATE,
+                      epochs=110,
+                      layers='heads')
+
+
+def detect_configuration(result_folder, weights_path):
+    '''Configure the MRCNN algorithme for the training'''
+    config = ship_config.ShipConfig()
+    config.display()
+
+    ship_model = model.MaskRCNN(mode="inference", config=config,
+                                model_dir=result_folder)
+
+    ship_model.load_weights(weights_path, by_name=True)
+
+    return ship_model, config
+
+
+def mask_analyse(mask):
+    '''Analyse the mrcnn mask result and return a string if exist'''
+    # We're treating all instances as one, so collapse the mask into one layer
+    mask = (np.sum(mask, -1, keepdims=True) >= 1)
+
+    if True in mask:
+        print("[INFO]: Ship(s) detected")
+        pixels = mask.T.flatten()
+        pixels = np.concatenate([[0], pixels, [0]])
+        ships = np.where(pixels[1:] != pixels[:-1])[0] + 1
+        ships[1::2] -= ships[::2]
+        return ' '.join(str(x) for x in ships)
+    else:
+        return None
+
+
+def ship_detection(images_file, images_dir, local_model):
+    '''Detect ships in image'''
+    dataset = pd.read_csv(images_file)
+    count = 1
+    results = list()
+    for image_name in dataset.iloc[:, 0].unique():
+        print(f"[INFO]: {count} / {len(dataset.iloc[:, 0].unique())} \n"
+              "The following image is analysed:", image_name)
+        image_path = images_dir + image_name
+        image = skimage.io.imread(image_path)
+        res_detection = local_model.detect([image], verbose=0)[0]["masks"]
+        mask_encoded = mask_analyse(res_detection)
+        results.append([image_name, mask_encoded])
+        count += 1
+
+    return results
+
+
+def export_result(list_results, folder):
+    '''Export results under csv format the results'''
+    dataset = pd.DataFrame(np.array(list_results), columns=['ImageId', 'EncodedPixels'])
```

Comment:
You should call your `DataFrame` objects `df` like the rest of the world does :)